### PR TITLE
feat(Gamut): switched ProgressBar back to taking in a 'style' prop

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Switched ProgressBar back to taking in a 'style' prop

Per discussions with @hmafzal, our preference (#749) is to keep atoms open to custom themes while we flesh out their usage. `ProgressBar` is still pretty new so we're going to want to keep it theme-happy.

### CODE REVIEW

File: packages/gamut/src/ProgressBar/index.tsx [hunk 1]
```diff
@@ -23,33 +21,43 @@ export type ProgressBarProps = {
    */
   percent: number;
 
-  theme: ProgressBarTheme;
+  style: ProgressBarStyle;
+};
+
+export type ProgressBarStyle = {
+  backgroundColor: string;
+  barColor: string;
```
Comment (hmafzal) [inline packages/gamut/src/ProgressBar/index.tsx:29]: Can't wait for the day we can type this as `barColor: GamutColor` :D

File: packages/gamut/src/ProgressBar/index.tsx [hunk 2]
```diff
@@ -23,33 +21,43 @@ export type ProgressBarProps = {
...
 
 export const ProgressBar: React.FC<ProgressBarProps> = ({
   className,
   large,
   minimumPercent = 0,
   percent,
-  theme,
+  style,
 }) => {
+  const { backgroundColor, barColor, fontColor } = style;
```
Comment (hmafzal) [inline packages/gamut/src/ProgressBar/index.tsx:40]: It might be valuable to specify some defaults here?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/ProgressBar/index.tsx:40]: Talked on Slack: Hasan suggested allowing _both_, where `theme` is the base and overrides can override it. That's pretty nifty. Let's do that!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.0.10-alpha.a4d33b.0<br/>@codecademy/gamut@7.1.4-alpha.a4d33b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ea863fe2e14f8e8129ee754--codecademy-styleguide.netlify.app>https://5ea863fe2e14f8e8129ee754--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ea863fe2e14f8e8129ee754>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.0.11-alpha.7091ce.0<br/>@codecademy/gamut@7.2.1-alpha.7091ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ea866a7fac361d982ef6371--codecademy-styleguide.netlify.app>https://5ea866a7fac361d982ef6371--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ea866a7fac361d982ef6371>Deploy Logs</a></p>

**Pull request merged**